### PR TITLE
Allow Task-async Tests to Fail

### LINKED ISSUE #406: Task-async Tests Cannot Fail
Sorry, this is all going to be in C# – my F# is not so good.

**Version** FsCheck v3.0.0-alpha1
**Related** FsCheck.Xunit v3.0.0-alpha1, xunit.core 2.3.0

Given the following xUnit.net test case:

```csharp
/// <summary>An example of the failure of async tests.</summary>
[Properties(QuietOnSuccess = true)]
public static class ExampleFixture
{
    [Property(DisplayName = "Irrespective of the exception thrown, the test passes.")]
    static async Task FailMe()
    {
        await Task.CompletedTask;
        Assert.False(true);
    }
}
```

...I'd expect it to fail. However, this test passes.

It looks like the issue is in [Testable][], testing for the success of a `Task`. I *think* that in `ContinueWith`, the `IsCompleted` property of a `Task` will always be `true`. What to check in order to know whether a `Task` has thrown is the `IsFaulted` property.

In LINQPad:

```csharp
await Task.FromException(new Exception("wow")).ContinueWith(t =>
{
  t.IsCompleted.Dump(); // True
  t.IsFaulted.Dump(); // True
  t.Exception.Dump(); // an aggregate exception
});

await Task.CompletedTask.ContinueWith(t =>
{
  t.IsCompleted.Dump(); // True
  t.IsFaulted.Dump(); // False
  t.Exception.Dump(); // null
});
```

[Testable]: https://github.com/fscheck/FsCheck/blob/ce793a4eedf3f1af4776345048341e1369ae8613/src/FsCheck/Testable.fs#L147-L149

### PR DESCRIPTION
**Pull request by chrisoverzero**
...by Checking Whether the Task Is Faulted

Fixes #406

### CODE REVIEW

**Comment by chrisoverzero** [comment]
I'm not at all sure that this is the optimal way to write this test. I only cribbed from other tests in the same file.

**Comment by Rattenkrieg** [comment]
I think we should behave defensively against canceled tasks too.

File: tests/FsCheck.Test/Property.fs [hunk 1]
```diff
@@ -156,7 +157,20 @@ module Property =
...
         a .|. b
-            
-        
-        
-
+    
+    let private unitGen =
+        Gen.constant ()
+    
+    [<Property>]
```
[the remark this answers is not in the document]
Comment (chrisoverzero) [reply tests/FsCheck.Test/Property.fs]: Y-Yes, this is better. I shouldn't write unit tests at 6:30PM.

**Comment by chrisoverzero** [comment]
@Rattenkrieg I can do that.

**Comment by kurtschelfthout** [comment]
Holy shit @chrisoverzero just like that you showed how to make Unquote work with the netcore test project. I'm unbelievably excited about this. Huge :+1: 

Also, great fix thanks much ;)

**Pull request merged**